dask/dataframe/core.py

```diff
@@ -7783,6 +7783,16 @@ def to_datetime(arg, meta=None, **kwargs):
             meta = meta_series_constructor(arg)([pd.Timestamp("2000", **tz_kwarg)])
             meta.index = meta.index.astype(arg.index.dtype)
             meta.index.name = arg.index.name
+    if PANDAS_GT_200 and "infer_datetime_format" in kwargs:
+        warnings.warn(
+            "The argument 'infer_datetime_format' is deprecated and will be removed in a future version. "
+            "A strict version of it is now the default, see "
+            "https://pandas.pydata.org/pdeps/0004-consistent-to-datetime-parsing.html. "
+            "You can safely remove this argument.",
+            UserWarning,
+        )
+        kwargs.pop("infer_datetime_format")
```

Comment:
`pop`ping it off here to avoid warnings being emitted in the `pd.datetime` calls in `map_partitions` below. This should be safe as `infer_datetime_format` no longer has an effect if it's specified in `pandas` 2.0

Reply:
But we still want the user to see the warning from `dask.dataframe`?

Reply:
Yeah, exactly. We still want to let users know that `infer_datetime_format` has been deprecated (hence emitting the warning) but we also don't want `pandas` itself to emit this warning a bunch of times in the `map_partitions(pd.to_datetime, ...)` call below. We just want the single warning when `dd.to_datetime` is called with `infer_datetime_format=` specified